apollo-router/src/plugins/rhai.rs

```diff
@@ -382,66 +468,48 @@ impl RhaiObjectSetterGetter for Extensions {
 
 #[derive(Clone, Debug)]
 pub(crate) struct RhaiContext {
+    headers: HeaderMap,
     context: Context,
 }
 
 impl RhaiContext {
-    fn new(context: Context) -> Self {
-        Self { context }
+    fn new(context: Context, headers: HeaderMap) -> Self {
+        Self { context, headers }
     }
-    /*
+
     fn get_headers(&mut self) -> Headers {
-        Headers(self.context.request.headers().clone())
+        Headers(self.headers.clone())
     }
     fn set_headers(&mut self, headers: Headers) {
-        *self.context.request.headers_mut() = headers.0;
+        self.headers = headers.0;
     }
-    */
-    /*
     fn get_entries(&mut self) -> Dynamic {
         to_dynamic(self.context.entries.clone()).unwrap()
     }
     fn set_entries(&mut self, entries: Dynamic) {
         self.context.entries = from_dynamic(&entries).unwrap();
     }
-    */
 }
 
 impl Rhai {
-    fn run_rhai_script(&self, function_name: &str, context: Context) -> Result<Context, String> {
-        let mut scope = Scope::new();
-        let response: RhaiContext = self
-            .engine
-            .call_fn(
-                &mut scope,
-                &self.ast,
-                function_name,
-                (RhaiContext::new(context),),
-            )
-            .map_err(|err| err.to_string())?;
-
-        Ok(response.context)
-    }
-
-    fn run_rhai_script_arc(
+    fn run_rhai_script(
         &self,
         function_name: &str,
         context: Context,
-    ) -> Result<Context, String> {
+        headers: HeaderMap,
+    ) -> Result<RhaiContext, String> {
         let mut scope = Scope::new();
-
-        let new_context = context.clone();
         let response: RhaiContext = self
             .engine
             .call_fn(
                 &mut scope,
                 &self.ast,
                 function_name,
-                (RhaiContext::new(new_context),),
+                (RhaiContext::new(context, headers),),
```

Comment:
I think this will have changed some examples won't it? Since, now the headers aren't coming from the request in the context but direct from the context.

Reply:
No it stays exactly the same because it's handled by setter and getter at the RHAI level and it didn't change there